src/engraving/infrastructure/eidregister.cpp

```diff
@@ -36,8 +37,13 @@ EID EIDRegister::newEID(ElementType type)
 
 void EIDRegister::registerItemEID(EID eid, EngravingObject* item)
 {
+#ifdef NDEBUG
+    UNUSED(eid);
+    UNUSED(item);
+#else
     bool inserted = m_register.emplace(eid.toUint64(), item).second;
     assert(inserted);
+#endif
```

Comment:
The insertion itself is needed in non-debug mode too!
```suggestion
    bool inserted = m_register.emplace(eid.toUint64(), item).second;
#ifdef NDEBUG
    UNUSED(inserted);
#else
    assert(inserted);
#endif
```